### PR TITLE
Pull all dependencies to the workspace

### PR DESCRIPTION
**Pull request by ggwpez**
Changes:
- Pull all dependencies to the workspace and add a CI check.
- Fixup the crates that had differing dependency aliases.
- Add CI check to keep deps in the workspace
- Add Taplo config for TOML formatting (follow up: add CI check)

This is a preliminary to updating to the new SDK version as we currently use conflicting aliases for a few packages while packages are not able to rename workspace deps, see https://github.com/rust-lang/cargo/issues/12546

This is a No-OP change, as can be verified with this:
```bash
git checkout oty-fix-dependencies
cargo tree -e features > oty.tree

git checkout origin/main
cargo tree -e features > main.tree

diff main.tree oty.tree
```

- [x] Does not require a CHANGELOG entry

### CODE REVIEW

**Review by bkchr — APPROVED** [review]
Ty!

File: system-parachains/encointer/Cargo.toml [hunk 1]
```diff
@@ -83,21 +79,21 @@ sp-genesis-builder = { workspace = true }
...
 cumulus-pallet-parachain-system = { features = [
-  "parameterized-consensus-hook",
+	"parameterized-consensus-hook",
 ], workspace = true }
 cumulus-pallet-xcm = { workspace = true }
 cumulus-pallet-xcmp-queue = { workspace = true }
 cumulus-primitives-aura = { workspace = true }
 cumulus-primitives-core = { workspace = true }
 cumulus-primitives-utility = { workspace = true }
-parachain-info = { package = "staging-parachain-info", default-features = false, version = "0.8.0" }
```
Comment (bkontur) [inline system-parachains/encointer/Cargo.toml:100]: I think the workspace dependencies feature is cool. The potential problem I see is with Encointer and its pallets, which use a different version of the Polkadot SDK. When we upgrade to a new version of the Polkadot SDK, Encointer is unlikely to compile. As a result, we won't be able to create a new release unless Encointer updates its pallets to the same Polkadot SDK version.

Therefore, it's questionable whether using workspace dependencies is a good idea for Encointer at this time. But let's give it a try and see what happens.

**Review by bkontur — APPROVED** [review]

File: system-parachains/encointer/Cargo.toml (hunk 1 above)
Comment (ggwpez) [reply system-parachains/encointer/Cargo.toml:100]: Okay i did not think about this. It basically means that we need to update the encointer pallets in advance to the version of the SDK that we plan to use next.

File: .cargo/config.toml [hunk 2]
```diff
@@ -7,27 +7,27 @@
...
-  "-Aclippy::bind_instead_of_map",       # styalistic
-  "-Aclippy::erasing_op",                # E.g. 0 * DOLLARS
-  "-Aclippy::eq_op",                     # In tests we test equality.
-  "-Aclippy::while_immutable_condition", # false positives
-  "-Aclippy::needless_option_as_deref",  # false positives
-  "-Aclippy::derivable_impls",           # false positives
-  "-Aclippy::stable_sort_primitive",     # prefer stable sort
-  "-Aclippy::extra-unused-type-parameters", # stylistic
+	"-Aclippy::all",
+	"-Aclippy::bind_instead_of_map",          # styalistic
```
Comment (ordian) [inline .cargo/config.toml:11]:
```suggestion
	"-Aclippy::bind_instead_of_map",          # stylistic
```

File: .cargo/config.toml [hunk 3]
```diff
@@ -7,27 +7,27 @@
...
+	"-Aclippy::borrowed-box",                 # Reasonable to fix this one
+	"-Aclippy::derivable_impls",              # false positives
+	"-Aclippy::eq_op",                        # In tests we test equality.
+	"-Aclippy::erasing_op",                   # E.g. 0 * DOLLARS
+	"-Aclippy::extra-unused-type-parameters", # stylistic
+	"-Aclippy::identity-op",                  # One case where we do 0 +
+	"-Aclippy::if-same-then-else",
+	"-Aclippy::needless_option_as_deref",     # false positives
+	"-Aclippy::nonminimal-bool",              # maybe
+	"-Aclippy::option-map-unit-fn",           # styalistic
```
Comment (ordian) [inline .cargo/config.toml:21]:
```suggestion
	"-Aclippy::option-map-unit-fn",           # stylistic
```

File: .cargo/config.toml [hunk 4]
```diff
@@ -7,27 +7,27 @@
...
+	"-Aclippy::stable_sort_primitive",        # prefer stable sort
+	"-Aclippy::too-many-arguments",           # (Turning this on would lead to)
+	"-Aclippy::type_complexity",              # raison d'etre
+	"-Aclippy::unit_arg",                     # styalistic.
```
Comment (ordian) [inline .cargo/config.toml:25]:
```suggestion
	"-Aclippy::unit_arg",                     # stylistic
```

**Review by ordian — APPROVED** [review]

**Review by bkontur — APPROVED** [review]

**Review by ordian — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Pull request merged**